### PR TITLE
[webpack]: add browserlist

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* Adds `browserlist` to `package.json` based on analytics from docs site (we don't have analytics on what devices people use the CMS on)
* Adds `browserslist-to-esbuild` to consume and convert the spec for `esbuild` – it doesn't do it natively see https://github.com/evanw/esbuild/issues/324

### Why is it needed?

* Because targets doesn't mean the spec has been implemented in browsers so this is a clearer definition of what we're trying to achieve.

### Related issue(s)/PR(s)

* discussed internally

### CODE REVIEW

File: packages/core/admin/package.json [hunk 1]
```diff
@@ -160,5 +161,10 @@
         ]
       }
     }
-  }
+  },
+  "browserslist": [
+    "last 3 major versions",
+    "Firefox ESR",
+    "last 2 Opera versions"
+  ]
```
Comment (gu-stav) [inline packages/core/admin/package.json:169]: I'd prefer if we could move this into a separate file `.browserslist` to have more visibility and to not overload the package.json so much. Wdyt?

Comment (joshuaellis) [reply packages/core/admin/package.json:169]: I have no strong opinions on this, but it needs to be shipped in the NPM package, looking at `npmignore` it _should be_ but I have more confidence in it being shipped in the `package.json` lol

**Review by gu-stav — COMMENTED** [review]
The changes look good to me and I am very happy to make the switch to relative targets. However: I think we should not merge this as a patch-release and wait for the next minor instead. Also, we should publish at least one experimental release and test a real-world app beforehand.

Does that work for you?

**Comment by Convly** [comment]
We just don't merge it right now and wait for the 4.5.6 to be released tomorrow. I'll switch the milestone

**Comment by Convly** [comment]
@joshuaellis @gu-stav 

Here's an exp for you :slightly_smiling_face: 

`npx create-strapi-app@0.0.0-210bce6ff069ed23ba25ab22c8ff3b8fc3e72973 my_app`

**Comment by joshuaellis** [comment]
Hey @Convly thanks for the experimental – it highlighted a mistake since now corrected...

Would you be able to do another experimental when you have a sec? 🙏🏼

**Comment by Convly** [comment]
:tipping_hand_man: `0.0.0-4d5f2fd152c65a79abbe842b715efad890469ca4`

**Comment by joshuaellis** [comment]
@gu-stav that experimental release works lovely FYI

**Review by gu-stav — COMMENTED** [review]
It works well for me - tested the app on different browsers (Edge latest, FF latest, Safari latest) and couldn't find any problems. ✅

File: packages/core/helper-plugin/webpack.config.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 const webpack = require('webpack');
 const { ESBuildMinifyPlugin } = require('esbuild-loader');
+const browserslistToEsbuild = require('browserslist-to-esbuild');
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/helper-plugin/webpack.config.js:3]: i do.

**Review by gu-stav — APPROVED** [review]

**Comment by laurenskling** [comment]
I think "not dead" should be in here. Aren't you including `last 3 major versions` of IE right now 🤷

**Comment by joshuaellis** [comment]
Possibly - feel free to submit a PR if you think it's incorrect! 😊

**Pull request merged**